redisinsight/ui/src/pages/home/components/form/TlsDetails.tsx

```diff
@@ -13,25 +13,58 @@ import {
 import cx from 'classnames'
 import { FormikProps } from 'formik'
 
+import { useDispatch } from 'react-redux'
 import { validateCertName, validateField } from 'uiSrc/utils'
+import PopoverDelete from 'uiSrc/pages/browser/components/popover-delete/PopoverDelete'
 
 import {
   ADD_NEW_CA_CERT,
   NO_CA_CERT
 } from 'uiSrc/pages/home/constants'
 import { DbConnectionInfo } from 'uiSrc/pages/home/interfaces'
-
+import { TelemetryEvent, sendEventTelemetry } from 'uiSrc/telemetry'
+import { deleteCaCertificateAction } from 'uiSrc/slices/instances/caCerts'
+import { deleteClientCertAction } from 'uiSrc/slices/instances/clientCerts'
 import styles from '../styles.module.scss'
 
+const suffix = '_tls_details'
+
 export interface Props {
   flexGroupClassName?: string
   flexItemClassName?: string
   formik: FormikProps<DbConnectionInfo>
   caCertificates?: { id: string; name: string }[]
   certificates?: { id: number; name: string }[]
 }
+
 const TlsDetails = (props: Props) => {
+  const dispatch = useDispatch()
   const { flexGroupClassName = '', flexItemClassName = '', formik, caCertificates, certificates } = props
+  const [activeCertId, setActiveCertId] = useState<string | null>(null)
+
+  const handleDeleteCaCert = (id: string) => {
+    dispatch(deleteCaCertificateAction(id, handleClickDeleteCert))
+  }
+
+  const handleDeleteClientCert = (id: string) => {
+    dispatch(deleteClientCertAction(id, handleClickDeleteCert))
+  }
+
+  const handleClickDeleteCert = (id: string) => {
+    sendEventTelemetry({
+      event: TelemetryEvent.CONFIG_DATABASES_CERTIFICATE_REMOVED,
+      eventData: {},
+    })
+    showPopover(id)
```

Comment:
I think showPopover we should call before deletion

Reply:
Removed this snippet as it's not required.

Reply:
why its not required?

Reply:
The popover component showPopover will show the popover. The telemetry event is sent after the certificate is removed, and we don't require showPopover because the popover is already available at that time. The Popover component's closePopover method closes the popover.

Reply:
yes, you are right, thanks